### DISCUSSION TITLE
Is scalaxb a good fit for large XML files?

### DISCUSSION THREAD

**Discussion by phdoerfler** [eed3si9n/scalaxb#658] [Q&A]
Unless I'm missing something, scalaxb relies on `NodeSeq` which is commonly implemented in a DOM-like fashion. With JAXB it is trivial to use StAX under the hood, allowing to process large XML files (200 MB+) with very little memory since the unmarshalled object graph is very small in comparison to the DOM overhead in terms of heap space.
What's the equivalent in scalaxb?

**Comment by eed3si9n** [comment]
I think scala-xml does have https://www.scala-lang.org/api/2.12.15/scala-xml/scala/xml/pull/index.html but as you mentioned scalaxb uses DOM-like `NodeSeq`, so yea I don't think it would work with large XML documents really well.
